lib/macho/macho_file.rb

```diff
@@ -170,26 +168,68 @@ def command(name)
 
     alias :[] :command
 
-    # Add a new load command to the Mach-O.
-    # @param lc [MachO::LoadCommand] the load command being added
+    # Replace a load command with another command in the Mach-O, preserving location.
+    # @param old_lc [MachO::LoadCommand] the load command being replaced
+    # @param new_lc [MachO::LoadCommand] the load command being added
     # @return [void]
     # @raise [MachO::HeaderPadError] if the new command exceeds the header pad buffer
-    # @note This is public, but methods like {#add_rpath} should be preferred.
-    def add_command(lc)
+    # @note This is public, but methods like {#dylib_id=} should be preferred.
+    def replace_command(old_lc, new_lc)
       context = LoadCommand::SerializationContext.context_for(self)
-      cmd_raw = lc.serialize(context)
+      cmd_raw = new_lc.serialize(context)
       new_sizeofcmds = sizeofcmds + cmd_raw.bytesize
-
       if header.class.bytesize + new_sizeofcmds > low_fileoff
```

Comment:
You're right, this is overly greedy. 

`new_sizeofcmds = sizeofcmds + cmd_raw.bytesize - old_cmd.cmdsize` should be correct, right?